File: app/src/main/java/com/example/s_tools/tools/retrofitcalls/InterceptorRetrofit.java
package com.example.s_tools.tools.retrofitcalls;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;

public class InterceptorRetrofit {
    public static OkHttpClient getInterceptor(){
        HttpLoggingInterceptor interceptor=new HttpLoggingInterceptor();
        interceptor.level(HttpLoggingInterceptor.Level.BODY);

        OkHttpClient okHttpClient = new OkHttpClient.Builder().addInterceptor(interceptor).build();
        return okHttpClient;
    }
}

File: app/src/main/java/com/example/s_tools/tools/retrofitcalls/InterceptorRetrofit.kt
package com.example.s_tools.tools.retrofitcalls

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor

object InterceptorRetrofit {
    @JvmStatic
    val interceptor: OkHttpClient
        get() {
            val interceptor = HttpLoggingInterceptor()
            interceptor.level = HttpLoggingInterceptor.Level.BODY
            return OkHttpClient.Builder().addInterceptor(interceptor).build()
        }
}